### PR TITLE
[develop] Trello-0HHIZ6m2: Added create category and get all

### PR DESCRIPTION
**Pull request by AndreySurzhan**

### CODE REVIEW

File: src/controllers/category.js [hunk 1]
```diff
@@ -1,93 +1,122 @@
-const Category = require('../models/category');
+//// Libs
+const _ = require('underscore');
+const Logger = require('../libs/log');
+const Q = require('q');
+const async = require('async');
```
Comment (AndreySurzhan) [inline src/controllers/category.js]: remove since it is not used

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
         "express": "4.0.0",
         "moment": "^2.14.1",
         "mongodb": "2.1.21",
-        "mongoose": "4.5.3",
+        "mongoose": "4.8.2",
```
Comment (AndreySurzhan) [inline package.json]: New in 4.8.0: update validators also run on $push and $addToSet

File: src/controllers/category.js [hunk 3]
```diff
@@ -1,93 +1,119 @@
...
-};
+//// Libs
+const Logger = require('../libs/log');
+const Q = require('q');
+/// Models
+const CategoryModel = require('../models/category');
+const ProjectModel = require('../models/project');
+/// Controllers
+/// Local variables
+let logger = Logger(module);
```
Comment (wildeastengineer) [inline src/controllers/category.js]: const

File: src/controllers/category.js [hunk 4]
```diff
@@ -1,93 +1,119 @@
...
 
+/**
+ * Categories controller.
+ * @class categoryController
+ */
 let categoryController = {
```
Comment (wildeastengineer) [inline src/controllers/category.js]: I suggest return value without variable creation

File: src/controllers/category.js [hunk 5]
```diff
@@ -1,93 +1,119 @@
...
+     * @param {(ObjectId|String)} data.id - Project's id
+     * @param {Object} data.category
+     * @param {String} data.category.name
+     * @param {String} data.category.categoryType
+     * @param {?(ObjectId|String)} data.category.parent
+     * @param {(ObjectId[]|String[])} data.currencies
+     *
+     * @returns {Promise<models/CategorySchema[]|Error>}
+     */
+    put: function (data) {
```
Comment (wildeastengineer) [inline src/controllers/category.js]: lets write it as `put(data) {`

File: src/controllers/category.js (hunk 4 above)
Comment (AndreySurzhan) [inline src/controllers/category.js]: @wildeastengineer what do you mean?
Exporting without variable creating?

Comment (wildeastengineer) [inline src/controllers/category.js]: @AndreySurzhan exactly. Like [here](https://github.com/wildeastengineer/MKeepAPI/pull/26/files)

**Comment by AndreySurzhan** [comment]
migration went well

File: src/controllers/category.js [hunk 6]
```diff
@@ -1,94 +1,118 @@
...
+     * @param {(ObjectId|String)} data.id - Project's id
+     * @param {Object} data.category
+     * @param {String} data.category.name
+     * @param {String} data.category.categoryType
+     * @param {?(ObjectId|String)} data.category.parent
+     * @param {(ObjectId[]|String[])} data.currencies
+     *
+     * @returns {Promise<models/CategorySchema[]|Error>}
+     */
+        put(data) {
```
Comment (wildeastengineer) [inline src/controllers/category.js]: Something wrong with indents

File: src/controllers/category.js [hunk 7]
```diff
@@ -1,94 +1,118 @@
...
+     * Get get all project categories
+     *
+     * @function
+     * @name getAll
+     * @memberof controllers/Currency
+     * @param {(ObjectId|String)} data.id - project id
+     *
+     * @returns {Promise<models/CategorySchema[]|Error>}
+     */
+        getAll(data) {
```
Comment (wildeastengineer) [inline src/controllers/category.js]: and here

**Comment by AndreySurzhan** [comment]
@wildeastengineer fixed. Please review

**Pull request merged**